drivers/sensor/weplatform/WeSensorsSDK.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief This is the main header file of the WE Sensors SDK.
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_WEPLATFORM_WESENSORSSDK_H_
+#define ZEPHYR_DRIVERS_SENSOR_WEPLATFORM_WESENSORSSDK_H_
+
+/**
+ * @brief SDK major version number.
+ */
+#define WE_SENSOR_SDK_MAJOR_VERSION 2
+
+/**
+ * @brief SDK minor version number.
+ */
+#define WE_SENSOR_SDK_MINOR_VERSION 2
+
+/**
+ * @brief Return code for successful operations.
+ */
+#define WE_SUCCESS 0
+
+/**
+ * @brief Return code for failed operations.
+ */
+#define WE_FAIL -1
+
+/**
+ * @brief Types of sensors supported by the WE Sensors SDK.
+ */
+typedef enum {
+	WE_HIDS, /* Humidity sensor */
+	WE_ITDS, /* Acceleration sensor */
+	WE_TIDS, /* Temperature sensor */
+	WE_PADS, /* Absolute pressure sensor */
+	WE_PDUS, /* Differential pressure sensor */
+	WE_SENSOR_TYPE_MAX
+} WE_sensorType_t;
```

Comment:
Remember that Zephyr prefers not `typedef`s and forbids the use of `CamelCase`.

Reply:
> Zephyr prefers not `typedef`s

Yes, this will be fixed along with all the other checkpatch warnings and errors.

> forbids the use of CamelCase

Ok. This hasn't been caught by the checkpatch script though. So is it a strict requirement for in-tree source files nevertheless?

Reply:
Yes, it is unfortunately. The `checkpatch` script doesn't catch everything, though it'd be great if it did at least catch this. cc @stephanosio.